web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -427,6 +481,76 @@ class AppConfig extends Component<Props, State> {
     return foundItem;
   };
 
+  // this runs on config update and when save is clicked but before the request is submitted
+  // on update it uses the errors from the liveconfig endpoint
+  // on save it's mostly used to find required field errors
+  mergeConfigGroupsAndValidationErrors = (
```

Comment:
this is the bulk of the logic. we take the `ConfigGroup[]` and merge in the `ConfigGroupItemValidationErrors[]` by matching the field names from each array.